_pytest/monkeypatch.py

```diff
@@ -32,62 +31,66 @@ def pytest_funcarg__monkeypatch(request):
     return mpatch
 
 
+def resolve(name):
+    # simplified from zope.dottedname
+    parts = name.split('.')
+
+    used = parts.pop(0)
+    found = __import__(used)
+    for part in parts:
+        used += '.' + part
+        try:
+            found = getattr(found, part)
+        except AttributeError:
```

Comment:
Suggestion: Remove nesting here using `else: continue`, such that tracebacks on Py3 don't contain a bunch of "exception happened while handling another exception"